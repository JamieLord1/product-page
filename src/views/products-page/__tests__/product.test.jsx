import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react'
import Product from '../product'

it('renders correctly', () => {
	const tree = renderer.create(<Product price={7.99} id={1} img="" name="" />).toJSON();
	expect(tree).toMatchSnapshot();
});

it('Buttons return correct data', () => {
	const { getByText } = render(<Product price={7.99} id={1} img="" name="" />)
	getByText('Remove').click()
	getByText('Qty')

	getByText('+').click()
	getByText('Qty 1')

	getByText('-').click()
	getByText('Qty')
});
