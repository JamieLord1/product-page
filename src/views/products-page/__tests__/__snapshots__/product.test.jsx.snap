// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="product-container"
>
  <img
    alt=""
    className="product-img"
    src=""
    width="100%"
  />
  <div
    className="product-text-container"
  >
    <h1
      className="product-name"
    >
      
    </h1>
    <label
      className="product-price"
    >
      Â£7.99
    </label>
  </div>
  <div
    className="product-qty-container"
  >
    <button
      className="ant-btn"
      onClick={[Function]}
      type="button"
    >
      <span
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        -
      </span>
    </button>
    <div
      className="product-qty-text-container"
    >
      <h2
        className="product-qty-text"
      >
        Qty 
      </h2>
      <button
        className="product-qty-remove-button"
        onClick={[Function]}
        type="button"
      >
        Remove
      </button>
    </div>
    <button
      className="ant-btn"
      onClick={[Function]}
      type="button"
    >
      <span
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        +
      </span>
    </button>
  </div>
</div>
`;
